CASC/PortfolioMode.cpp

```diff
@@ -378,7 +380,34 @@ bool PortfolioMode::runSchedule(Schedule& schedule)
   PortfolioSliceExecutor executor(this);
   ScheduleExecutor sched(&policy, &executor);
 
-  return sched.run(schedule);
+  bool result = sched.run(schedule);
+
+  //All children have been killed. Now safe to print proof
+  if(result && !env.options->printProofToFile()){
+    vstring fname = env.options->problemName() + "-vampire.proof";    
+    BYPASSING_ALLOCATOR; 
+    
+    istream* input=new ifstream(fname.c_str());
+
+    bool openSucceeded = !input->fail();
+
+    if (openSucceeded) {
+      env.beginOutput();
+      env.out() << input->rdbuf();
+      env.endOutput();
+    }
+
+    delete static_cast<ifstream*>(input);
```

Comment:
Is the static cast really necessary? I would expect polymorphism to take care of this. 

Also, can't you declare `input` as `ifstream*` in the first place?

Or even: Can't you declare `input` as `ifstream` and rely on RAII for the cleanup?

Reply:
I copied this code from `UIHelper::getIputProblem`

```
  istream* input;
  if (inputFile=="") {
    input=&cin;
  } else {
    // CAREFUL: this might not be enough if the ifstream (re)allocates while being operated
    BYPASSING_ALLOCATOR; 
    
    input=new ifstream(inputFile.c_str());
    if (input->fail()) {
      USER_ERROR("Cannot open problem file: "+inputFile);
    }
  }
```
But on consideration, I think the existing code is suboptimal. I have refactored to declare as `ifstream` and use RAll